ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java

```diff
@@ -761,19 +827,78 @@ private static Instant asInstant(final Expression other) {
   }
 
   private enum ComparisonTarget {
-    KEYCOL,
     WINDOWSTART,
     WINDOWEND
   }
 
-  private static Map<ComparisonTarget, List<ComparisonExpression>> extractComparisons(
+  private static class KeyAndWindowBounds {
```

Comment:
What's the benefit of having this class versus adding one more entry in the `ComparisonTarget`? Seems a lot of work for achieving the same functionality?

Reply:
InPredicate isn't a ComparisonExpression.  Here, I can have a different type for each target type.

Alternatively, I could have change the map to point to an expression and casted back, but that's not a great practice.